cedar-policy-core/src/ast/entity.rs

```diff
@@ -346,6 +355,20 @@ impl Entity {
         })
     }
 
+    /// Create a new `Entity` with this UID, ancestors, and an empty set of attributes
+    /// Since it lacks any attributes, this method returns `Self` instead of `Result<Self>`
+    pub fn new_empty_attrs(
+        uid: EntityUID,
+        ancestors: HashSet<EntityUID>,
+        extensions: &Extensions<'_>,
+    ) -> Self {
+        // Safe as the attributes hashmap is empty, and `Self::new`
+        // can only error when evaluating an attribute expression
+        // PANIC SAFETY (see above)
+        #[allow(clippy::unwrap_used)]
+        Self::new(uid, HashMap::new(), ancestors, extensions).unwrap()
```

Comment:
Relying on this invariant for panic safety seems sketchy to me, I could imagine a future PR adding a way `Entities::new()` could error, but not updating this code or the referenced error docs.

I think we could avoid the unwrap by relying on Core's `Entity::new_with_attr_partial_value()` which does not error.  (You can pass an empty attr map to it.)